test/integration/options/test_configure_options.py

```diff
@@ -123,3 +123,32 @@ class Pkg(ConanFile):
         client.save({"conanfile.py": conanfile})
         client.run(f"create . --name=pkg --version=0.1")
         self.assertIn("Package 'da39a3ee5e6b4b0d3255bfef95601890afd80709' created", client.out)
+
+
+def test_possible_values_based_on_version():
+    tc = TestClient(light=True)
+    tc.save({"conanfile.py": textwrap.dedent("""
+    from conan import ConanFile
+    from conan.tools.scm import Version
+
+    class Pkg(ConanFile):
+        name = "foo"
+        options = {"my_option": ["foo", "v1", "v2"]}
+        default_options = {"my_option": "foo"}
+        def config_options(self):
+            if Version(self.version) < "2.0":
+                self.options.my_option.constrain_possible_values(["foo", "v1"])
+            else:
+                self.options.my_option.constrain_possible_values(["foo", "v2"])
+
+    """)})
```

Comment:
I was thinking about something slightly different:

```python
        def config_options(self):
            # foo is available for all the versions, so there is no need to add it here
            self.options.my_option.version_constrain(["v1"], "< 2.0")
            self.options.my_option.version_constrain(["v2"], ">= 2.0")
```

Could it be feasible?

Reply:
I think it could introduce even more confussion - But the fact that you are suggesting this maybe tells me that the feature is not 100% clear/consistent 🤔